src/snek5000/output/print_stdout.py

```diff
@@ -19,96 +21,166 @@ class PrintStdOut:
-                (.+)           # 3: Capture C: any char
-    """,
+                (?P<CFL>\S+)   # 3: Capture C: any non-whitespace char
+            """
+        # TODO: add undocumented params.nek.pressure.solver parameter
+        # See Line 445 in Nek5000/core/reader_par.f
+        pattern_pressure = rf"""^\ +
+                (?P<it2>\d+)
+                .*
+                PRES\ {pressure_solver}    # For all lines containing a string like PRES gmres
+                \s+
+                (?P<pres_it>\d+)
+                \s+
+                (?P<pres_div>\S+)
+                \s+
+                (?P<pres_div0>\S+)
+                \s+
+                (?P<pres_tol>\S+)
+                \s+
+                (?P<pres_etime>\S+)
+                \s+
+                (?P<pres_etime1>\S+)
+            """
+
+        expr = re.compile(
+            f"({pattern_step})|({pattern_pressure})",
             re.VERBOSE | re.MULTILINE,
         )
-        return pattern.findall(self.text)
+
+        keys_all = tuple(expr.groupindex)
+        # Divide into two tuples at "it2"
+        idx = keys_all.index("it2")
+        keys_step, keys_pressure = keys_all[:idx], keys_all[idx:]
+
+        matches = tuple(expr.finditer(self.text))
+
+        def make_dict(keys):
+            return {key: [m[key] for m in matches] for key in keys}
+
+        def make_df(keys):
+            return (
+                pd.DataFrame.from_dict(make_dict(keys))
+                .dropna()
+                .astype({key: int if key == "it" else float for key in keys})
+            )
+
+        df_step = make_df(keys_step).set_index("it")
+        df_pressure = (
+            make_df(keys_pressure).rename(columns={"it2": "it"}).set_index("it")
+        )
+
+        self.data = df_step.join(df_pressure)
+        self._data_modif_time = path_file_time
+        return self.data
 
     @property
     def dt(self):
         """Extract time step dt over the course of a simulation."""
-        # Alternate implementations
-        # 1. awkup
-        # dts = %awk -F, -e '/Step/{print $3}' {self.file}
-        #
-        # 2. Basic regex
-        # steps = re.findall("Step.*DT.*", self.text)
-        # try:
-        #     dt_strings = [step.split(',')[2] for step in steps if step]
-        #     dts = [float(s.split("= ")[1]) for s in dt_strings if s]
-        # except IndexError:
-        #     print(steps)
-        dt = [float(field[2]) for field in self.text_steps]
-        return np.array(dt)
+        warn(
+            "The property `dt` is deprecated. Use `load()` instead.", DeprecationWarning
+        )
+        return self.data.dt
+
+    def estimate_clock_times(self):
+        """Computed estimated walltime / clock time to complete the simulation
+        and clock time per time step
+
+        """
+        last_data = self.load().tail(1)
+
+        # FIXME: ctime is not creation time?
+        first_timestamp = datetime.fromtimestamp(self.path_file.stat().st_ctime)
```

Comment:
A better approach is needed to find out when the simulation began. Perhaps look at when SESSION.NAME was changed. [File creation time is not available in Linux](https://stackoverflow.com/a/39501288)

Reply:
Can't we just write this time in the log file?

Reply:
It would indeed be much better not to rely on the modification time of the file. We anyway need the clock time of some time steps so we can forget about these variables here and add another dedicated file in another PR (see #137).

Reply:
Removed the method.